setup.py

```diff
@@ -43,50 +36,82 @@ def build_extension(self, ext):
         except ext_errors:
             raise BuildFailed()
 
+
 class ve_build_py(build_py):
     def run(self, *args, **kwds):
         if not try_krb:
             print("libgssapi_krb5 not available, skipping kerberos module")
         build_py.run(self, *args, **kwds)
 
+
 try:
     libgssapi_krb5 = ctypes.CDLL("libgssapi_krb5.so")
-    defines = [("HAVE_GSS_SET_CRED_OPTION", hasattr(libgssapi_krb5,
-                                                    "gss_set_cred_option")),
-               ("HAVE_GSSSPI_SET_CRED_OPTION", hasattr(libgssapi_krb5,
-                                                       "gssspi_set_cred_option"))]
-    lw_krb_module = Extension('pike.kerberos',
-                              ["pykerb/base64.c",
-                               "pykerb/kerberosbasic.c",
-                               "pykerb/kerberos.c",
-                               "pykerb/kerberosgss.c",
-                               "pykerb/kerberospw.c"],
-                              libraries=['gssapi_krb5'],
-                              define_macros=defines)
+    defines = [
+        ("HAVE_GSS_SET_CRED_OPTION", hasattr(libgssapi_krb5, "gss_set_cred_option")),
+        (
+            "HAVE_GSSSPI_SET_CRED_OPTION",
+            hasattr(libgssapi_krb5, "gssspi_set_cred_option"),
+        ),
+    ]
+    lw_krb_module = Extension(
+        "pike.kerberos",
+        [
+            "pykerb/base64.c",
+            "pykerb/kerberosbasic.c",
+            "pykerb/kerberos.c",
+            "pykerb/kerberosgss.c",
+            "pykerb/kerberospw.c",
+        ],
+        libraries=["gssapi_krb5"],
+        define_macros=defines,
+    )
 except OSError:
     try_krb = False
 
+
 def pike_suite():
-    return unittest.defaultTestLoader.discover('pike/test', pattern='*.py')
+    return unittest.defaultTestLoader.discover("pike/test", pattern="*.py")
+
 
 def run_setup(with_extensions):
     ext_modules = []
-    cmdclass = { "build_py": ve_build_py }
+    cmdclass = {"build_py": ve_build_py}
     if with_extensions:
         ext_modules.append(lw_krb_module)
         cmdclass = dict(cmdclass, build_ext=ve_build_ext)
-    setup(name='Pike',
-          version=__version__,
-          description='Pure python SMB client',
-          author='Brian Koropoff',
-          author_email='Brian.Koropoff@emc.com',
-          url='https://github.com/emc-isilon/pike',
-          packages=['pike', 'pike.test'],
-          install_requires=['pycryptodomex'],
-          ext_modules=ext_modules,
-          test_suite='setup.pike_suite',
-          cmdclass=cmdclass,
-          )
+    setup(
+        name="pike-smb2",
+        use_scm_version=True,
+        setup_requires=["setuptools_scm"],
+        description="Pure python SMB client",
```

Comment:
I recommend pulling in the nicely written markdown `README.md`.

    _HERED = os.path.abspath(os.path.dirname(__file__))
    _README = os.path.join(_HERED, 'README.md')
    ...
    # Get the long description from the README.md file
    with open(_README, encoding='utf-8') as f_:
        long_description = f_.read()

    ...
    setup(
       ...
               long_description_content_type='text/markdown',
              long_description=long_description,